File: C++/Print Bracket Number.cpp
class Solution { 
  public:
    vector<int> bracketNumbers(string str) {
        stack<pair<char, int>> st;
        vector<int> ans;
        int maxi = 0;
        for(auto it: str){
            if(it == '('){
                maxi++;
                st.push({'(', maxi});
                ans.push_back(maxi);
            }
            else if(it == ')'){
                ans.push_back(st.top().second);
                st.pop();
            }
        }
        return ans;
    }
};

File: JAVA/Print Bracket Number.java
class Solution {
    ArrayList<Integer> bracketNumbers(String str) {
    Stack<Integer> stack = new Stack<>();
    ArrayList<Integer> list= new ArrayList();
    int count=0;
    for(int i=0;i<str.length();i++){
       if(str.charAt(i)=='('){
           count++;
           stack.push(count);
           list.add(count);
       } 
        if(str.charAt(i)==')'){
            list.add(stack.pop());
        }
    }
     return list;
    }
}
